### PR TITLE
Improve transaction handling

### PR DESCRIPTION
**Pull request by kimchitran-cap**

### CODE REVIEW

File: src/test/java/eu/merloteducation/contractorchestrator/ContractStorageServiceTest.java [hunk 1]
```diff
@@ -1135,6 +1227,8 @@ void transitionDataDeliveryContractPurge() throws IOException {
         representedOrgaIds.add(consumer);
         representedOrgaIds.add(provider);
         DataDeliveryContractTemplate template = new DataDeliveryContractTemplate(dataDeliveryContract, false);
+        template.setServiceContractProvisioning(new DataDeliveryProvisioning()); // reset provisioning
```
Comment (M-Busk) [inline src/test/java/eu/merloteducation/contractorchestrator/ContractStorageServiceTest.java:1230]: why is this saved after each step?

File: src/test/java/eu/merloteducation/contractorchestrator/ContractStorageServiceTest.java [hunk 2]
```diff
@@ -1322,6 +1422,54 @@ void addContractAttachmentsInDraft() throws IOException {
...
+        });
+        assertNotNull(check1);
+        assertFalse(check1.getAttachments().contains("myFile.pdf"));
+
+        Exception thrownEx = null;
+        try {
+            transactionTemplate.execute(status -> {
+                try {
+                    contractStorageService.addContractAttachment(template.getId(), new byte[]{}, "myFile.pdf", "authToken");
+                } catch (IOException ex) {
```
Comment (M-Busk) [inline src/test/java/eu/merloteducation/contractorchestrator/ContractStorageServiceTest.java:1451]: can this ioexception ever be thrown? it is caught in the service

Comment (kimchitran-cap) [reply src/test/java/eu/merloteducation/contractorchestrator/ContractStorageServiceTest.java]: The service method had an unnecessary "throws IOException". This has been dealt with.

File: src/test/java/eu/merloteducation/contractorchestrator/ContractStorageServiceTest.java (hunk 1 above)
Comment (kimchitran-cap) [reply src/test/java/eu/merloteducation/contractorchestrator/ContractStorageServiceTest.java:1230]: The template is being saved in several test cases, not just in one. These test cases have been revisited and if needed revised.

**Pull request merged**